plugins/autocomplete/plugin.js

```diff
@@ -1336,28 +1342,23 @@
 	 * ```javascript
 	 *	// Returns (through its callback) the suggestions for the current query.
 	 *	// Note: the itemsArray variable is our example "database".
-	 *	function dataCallback( query, range, callback ) {
+	 *	function dataCallback( options ) {
 	 *		// Simple search.
 	 *		// Filter the entire items array so only the items that start
 	 *		// with the query remain.
 	 *		var suggestions = itemsArray.filter( function( item ) {
-	 *			return item.name.indexOf( query ) === 0;
+	 *			return item.name.indexOf( options.query ) === 0;
 	 *		} );
 	 *
 	 *		// Note - the callback function can also be executed asynchronously
 	 *		// so dataCallback can do an XHR requests or use any other asynchronous API.
-	 *		callback( suggestions );
+	 *		options.callback( suggestions );
 	 *	}
 	 *
 	 * ```
 	 *
 	 * @method dataCallback
-	 * @param {String} query The query string that was accepted by the `textTestCallback`.
-	 * @param {CKEDITOR.dom.range} range The range in the DOM where the query text is.
-	 * @param {Function} callback The callback which should be executed with the matched data.
-	 * @param {CKEDITOR.plugins.autocomplete.model.item[]} callback.data The suggestion data that should be
-	 * displayed in the autocomplete view for a given query. The data items should implement the
-	 * {@link CKEDITOR.plugins.autocomplete.model.item} interface.
+	 * @param {CKEDITOR.plugins.autocomplete.dataCallbackDefinition} dataCallbackDefinition
```

Comment:
`dataCallbackDefinition` does not sound very helpful. If you look at the following usage sample:

```js
// Returns (through its callback) the suggestions for the current query.
// Note: the itemsArray variable is our example "database".
function dataCallback( options ) {
    // Simple search.
    // Filter the entire items array so only the items that start
    // with the query remain.
    var suggestions = itemsArray.filter( function( item ) {
        return item.name.indexOf( options.query ) === 0;
    } );

    // Note - the callback function can also be executed asynchronously
    // so dataCallback can do an XHR requests or use any other asynchronous API.
    options.callback( suggestions );
}
```

What is now `dataCallbackDefinition` acts as object with information about match, based on which proper data should be returned. The only thing that does not fit there is that it contains a `callback` which is not related to match itself.

So how about extracting callback as a second argument, and rename `dataCallbackDefinition` to describe better what it conveys?